util/src/main/java/org/killbill/billing/util/entity/dao/SearchQuery.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020-2024 Equinix, Inc
+ * Copyright 2014-2024 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.util.entity.dao;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.annotation.Nullable;
+
+public class SearchQuery {
+
+    public static final String SEARCH_QUERY_MARKER = "_q=1&";
+
+    private final Map<String, Object> bindMap = new HashMap<>();
+    private final List<SearchAttribute> attributes = new LinkedList<>();
+    private final SqlOperator sqlOperator;
```

Comment:
This confused me first, as the `SqlOperator` is used for different purposes. (just a comment)

Reply:
Yup. I had different objects at first but then just merged them...